assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/factory.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Factory module to create task based convertor classes."""
+
+from abc import ABC, abstractmethod
+from azureml.model.mgmt.processors.pyfunc.config import (
+    SupportedVisionTasks, SupportedTasks
+)
+from azureml.model.mgmt.utils.logging_utils import get_logger
+from .convertors import (
+    VisionMLFlowConvertor, CLIPMLFlowConvertor
+)
+
+
+logger = get_logger(__name__)
+
+
+def get_mlflow_convertor(model_dir, output_dir, temp_dir, translate_params):
+    """Instantiate and return PyFunc MLflow convertor."""
+    task = translate_params["task"]
+    if SupportedVisionTasks.has_value(task):
+        return VisionMLflowConvertorFactory.create_mlflow_convertor(model_dir, output_dir, temp_dir, translate_params)
+    elif task == SupportedTasks.ZERO_SHOT_IMAGE_CLASSIFICATION.value:
+        return CLIPMLflowConvertorFactory.create_mlflow_convertor(model_dir, output_dir, temp_dir, translate_params)
```

Comment:
Now, we can have CLIP_HF_Ml..... factory depending on model source and task.

Reply:
Will leave it as is for now. Maybe we can edit it if there is clip model from another source added later.